akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorRetentionSpec.scala

```diff
@@ -127,6 +127,42 @@ class EventSourcedBehaviorRetentionSpec
   val pidCounter = new AtomicInteger(0)
   private def nextPid(): PersistenceId = PersistenceId.ofUniqueId(s"c${pidCounter.incrementAndGet()}")
 
+  "EventSourcedBehavior with predicate" must {
+    "delete events after snapshot" in {
+      val eventProbe = TestProbe[Try[EventSourcedSignal]]()
+      val pid = nextPid()
+      val snapshotSignalProbe = TestProbe[WrappedSignal]()
+      val alwaysSnapshot: Behavior[Command] =
+        Behaviors.setup { ctx =>
+          counter(
+            ctx,
+            pid,
+            snapshotSignalProbe = Some(snapshotSignalProbe.ref),
+            eventSignalProbe = Some(eventProbe.ref)).snapshotWhen((_, _, _) => true, deleteEventsOnSnapshot = true)
+        }
+      val c = spawn(alwaysSnapshot)
+      val replyProbe = TestProbe[State]()
+
+      c ! Increment
+      snapshotSignalProbe.expectSnapshotCompleted(1)
+      eventProbe.expectMessageType[Success[DeleteEventsCompleted]].value.toSequenceNr shouldEqual 1
+      c ! GetValue(replyProbe.ref)
+      replyProbe.expectMessage(State(1, Vector(0)))
+      c ! StopIt
+      val watchProbe = TestProbe()
+      watchProbe.expectTerminated(c)
+
+      val probe = TestProbe[(State, Event)]()
+      val c2 = spawn(Behaviors.setup[Command](ctx => counter(ctx, pid, Some(probe.ref))))
+      // state should be rebuilt from snapshot, no events replayed
+      // Fails as snapshot is async (i think)
```

Comment:
is this a question or a possibly flaky test?

Reply:
Comment could be removed.

Reply:
Comment still here though